osu.Game/Online/API/Requests/Responses/APICommentsController.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using Newtonsoft.Json;
+using osu.Game.Users;
+using System.Collections.Generic;
+
+namespace osu.Game.Online.API.Requests.Responses
+{
+    public class APICommentsController
+    {
+        private List<Comment> comments;
+
+        [JsonProperty(@"comments")]
+        public List<Comment> Comments
+        {
+            get => comments;
+            set
+            {
+                comments = value;
+                comments.ForEach(child =>
+                {
+                    if (child.ParentId != null)
+                    {
+                        comments.ForEach(parent =>
+                        {
+                            if (parent.Id == child.ParentId)
+                            {
+                                parent.ChildComments.Add(child);
+                                child.ParentComment = parent;
+                            }
+                        });
+                    }
+                });
```

Comment:
I think you could take this down from being quadratic to at least quasilinear (`n log n`) with some dictionaries. And this could matter for long comment threads, too.